File: projectEuler/10/10.cpp
#include <iostream>
#include <cmath>

using namespace std;

bool isPrime(int n) {
    if (n == 1) return false;
    if (n == 2) return true;
    if (n % 2 == 0) return false;
    for (int i = 3; i <= sqrt(n); i += 2) {
        if (n % i == 0) return false;
    }
    return true;
}

long long sumPrimes(int n) {
    long long sum = 0;
    for (int i = 2; i < n; i++) {
        if (isPrime(i)) {
            sum += i;
        }
    }
    return sum;
}

int main() {
    cout << sumPrimes(2000000) << endl;
    return 0;
}

File: projectEuler/10/10.py
#Find the sum of all the primes below two million.

def isPrime(n):
    if n == 1:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    for i in range(3, int(n**0.5)+1, 2):
        if n % i == 0:
            return False
    return True

def sumPrimes(n):
    sum = 0
    for i in range(2, n):
        if isPrime(i):
            sum += i
    return sum

print(sumPrimes(2000000))

